src/library/trackset/setlogfeature.cpp

```diff
@@ -105,6 +108,46 @@ SetlogFeature::~SetlogFeature() {
     deleteAllUnlockedPlaylistsWithFewerTracks();
 }
 
+
+void SetlogFeature::addPlaylistIfCmdArg() {
+    const QString playlistFile = CmdlineArgs::Instance().getPlaylistFilePath();
```

Comment:
Even if it is convenient, we really should not mix command-line arguments into some unrelated feature. As a proof-of-concept it's of course fine, but we should definitely move this elsewhere before merging.

My gut feeling would actually be to have the argument handling itself in `main.cpp` and from there to call out to CoreServices' `trackCollectionManager`, which should also give you a path to the playlist DAO. Check out `WTrackMenu` for how this is done for the "Import as Playlist" context menu entries:

https://github.com/mixxxdj/mixxx/blob/d961d79e95404130f43a24768f5538f3db593c1a/src/widget/wtrackmenu.cpp#L1218-L1220